packages/@internationalized/number/src/NumberParser.ts

```diff
@@ -215,6 +215,10 @@ class NumberParserImpl {
       }
     }
 
+    if (this.symbols.group && value.includes("'")) {
+      value = replaceAll(value, "'", this.symbols.group);
```

Comment:
this seems like an assumption that might break in other locales, formats, etc. maybe only do it if the group symbol is the actual unicode quotation mark we expect?

can you also add a comment explaining this?